test/hummingbot/strategy/avellaneda_market_making/test_avellaneda_market_making.py

```diff
@@ -0,0 +1,1145 @@
+
+        max_spread_around_reserved_price = max_spread * (2 - self.ira) + min_spread * self.ira
+        expected_kappa = expected_gamma / (Decimal.exp((max_spread_around_reserved_price * expected_gamma - (vol * expected_gamma) ** 2) / 2) - 1)
+
+        q_where_to_decay_order_amount = self.strategy.calculate_target_inventory() / (self.ira * Decimal.ln(Decimal("10")))
+        expected_eta = s_decimal_one / q_where_to_decay_order_amount
+
+        self.strategy.recalculate_parameters()
+        self.assertAlmostEqual(expected_gamma, self.strategy.gamma, 5)
+        self.assertAlmostEqual(expected_kappa, self.strategy.kappa, 5)
+        self.assertAlmostEqual(expected_eta, self.strategy.eta, 5)
+
+        # Simulate close to _inventory_target_base_pct
+        self.market.set_balance("COINALPHA", 5)
+        self.market.set_balance("HBOT", 500)
+
+        q = (self.market.get_balance(self.base_asset) - self.strategy.calculate_target_inventory()) * self.strategy.q_adjustment_factor
+        vol = self.strategy.get_volatility()
+        min_spread, max_spread = self.strategy._get_min_and_max_spread()
+
+        # TODO: Test for expected_gamma = self.ira * (max_spread * (2-self.ira) / self.ira + min_spread) / (vol ** 2)
+
+    def test_calculate_reserved_price_and_optimal_spread(self):
+        # Test (1) Low volatility, Default min_spread = Decimal(0.2)
+        # Simulate low volatility
+        self.simulate_low_volatility(self.strategy)
+
+        # Prepare parameters for calculation
+        self.strategy.recalculate_parameters()
+
+        price = self.strategy.get_price()
+        q = (self.market.get_balance(self.base_asset) - self.strategy.calculate_target_inventory()) * self.strategy.q_adjustment_factor
+        vol = self.strategy.get_volatility()
+        mid_price_variance = vol ** 2
+
+        time_left_fraction = Decimal("1")
+        expected_reserved_price = price - (q * self.strategy.gamma * mid_price_variance * time_left_fraction)
+        expected_optimal_spread = self.strategy.gamma * mid_price_variance * time_left_fraction + 2 * Decimal(
+            1 + self.strategy.gamma / self.strategy.kappa).ln() / self.strategy.gamma
+        expected_optimal_ask = expected_reserved_price + expected_optimal_spread / 2
+        expected_optimal_bid = expected_reserved_price - expected_optimal_spread / 2
+
+        self.strategy.calculate_reserved_price_and_optimal_spread()
+
+        # Check reserved_price, optimal_ask and optimal_bid
+        self.assertAlmostEqual(expected_reserved_price, self.strategy.reserved_price, 2)
+        self.assertAlmostEqual(expected_optimal_spread, self.strategy.optimal_spread, 2)
+        self.assertAlmostEqual(expected_optimal_ask, self.strategy.optimal_ask, 1)
+        self.assertAlmostEqual(expected_optimal_bid, self.strategy.optimal_bid, 1)
+
+        # TODO: Test for different paths optimal_ask and optimal_bid. See AvellanedaMM line 631-648.
```

Comment:
Is this TODO still pending resolution?

Reply:
See comment [here](https://github.com/CoinAlpha/hummingbot/pull/3483#discussion_r655374449).